index.d.ts

```diff
@@ -0,0 +1,108 @@
+declare module 'dicom-parser' {
+ 
+  export type ByteArray = Uint8Array | Buffer;
+
+  export interface Element {
+    tag: string;
+    vr?: string;
+    length: number;
+    dataOffset: number;
+    items?: Element[];
+    dataSet?: DataSet;
+    parser?: ByteArrayParser;
+  }
+
+  export interface DataSet {
+    byteArray: ByteArray;
+    byteArrayParser : ByteArrayParser;
+    elements: {
+      [key: string]: Element;
+    };
+    warnings: string[];
```

Comment:
I believe `warnings` are optional, right ?

Reply:
I think it's always present. The `DataSet` gets its `warning` field set from the `ByteStream` [after it's instantiated](https://github.com/cornerstonejs/dicomParser/blob/374ad3d20a92bd2a7ec56d90dfce2fb2815fee13/src/parseDicom.js#L124), and `ByteStream` [always has](https://github.com/cornerstonejs/dicomParser/blob/374ad3d20a92bd2a7ec56d90dfce2fb2815fee13/src/byteStream.js#L45) a `warnings` value.

Please let me know if I'm missing something though.